federation/pkg/dnsprovider/providers/google/clouddns/zones.go

```diff
@@ -42,6 +44,36 @@ func (zones Zones) List() ([]dnsprovider.Zone, error) {
 	return zoneList, nil
 }
 
+func (zones Zones) Get(dnsZoneName string, dnsZoneID string) (dnsprovider.Zone, error) {
+	if len(dnsZoneID) > 0 {
```

Comment:
This logic seems complex and unnecessary.  For a given cloud provider, is it not the case that one of dnsZoneName or dnsZoneId should always be empty, and the other always non-empty?  Also, same comment above regarding the unclear and duplicated trim logic.